src/class/cdc/cdc_device.c

```diff
@@ -357,8 +353,21 @@ bool cdcd_control_request(uint8_t rhport, tusb_control_request_t const * request
       bool const dtr = tu_bit_test(request->wValue, 0);
       bool const rts = tu_bit_test(request->wValue, 1);
 
+#if CFG_TUD_CDC_CLEAR_AT_CONNECTION
+      // DTE connected event (if DTE supports DTR bit)
+      if ( dtr && !tu_bit_test(p_cdc->line_state, 0) )
+      {
+        // Clear not transmitted data
+        usbd_edpt_xfer_abort(rhport, p_cdc->ep_in);
+        tu_fifo_clear(&p_cdc->tx_ff);
+      }
+#endif
+
       p_cdc->line_state = (uint8_t) request->wValue;
 
+      // Disable fifo overwriting if DTR bit is set
+      tu_fifo_change_mode(&p_cdc->tx_ff, (dtr?false:true));
```

Comment:
Thanks a lot. I was struggling a lot with finding a good name for it. Like your idea! 😉 Also I wasn't 100% sure if we really need this function. We could just change the overwritable bit within tx_ff from cdc_device.c. Don't see a real problem here if we do not lock the fifo. But like now it might be safe for sure.